### PR TITLE
oe_vectortranslate tests, docs and examples

### LINKED ISSUE #58: Document how to use osmconf_ini argument
Heads-up @agila5 I think we had docs in the previous version, does it work the same?

### LINKED ISSUE #80: Check for existing attributes
The vectortranslate operation is expensive so I would add a check here

https://github.com/ITSLeeds/osmextractr/blob/6c0ab50aeecff02cb3b96b26a71f5c0c8b6bc11a/R/get.R#L110-L112

to check if the input `extra_attributes` are already present in the `.gpkg` file (maybe with ogrinfo -fields=YES) and, in that case, set `force_vectortranslate = FALSE`

### LINKED ISSUE #107: Display extra fields stored in "other_tags"
I think it could be a good idea creating a function for displaying the extra fields stored in the column "other_tags". For example here: 

``` r
library(sf)
osmextract::oe_get("Andorra")[9]
other_tags
#> 1   "maxspeed"=>"30","old_ref"=>"N 22","oneway"=>"yes","ref"=>"N 22"
#> 3   "maxspeed"=>"30","old_ref"=>"CG-2","oneway"=>"yes","ref"=>"CG-2"
#> 4   "junction"=>"roundabout","ref"=>"CG-2"
#> 5   "maxspeed"=>"30","old_ref"=>"CG-2","oneway"=>"yes","ref"=>"CG-2"
#> 6   "maxspeed"=>"30","old_ref"=>"CG-2","oneway"=>"yes","ref"=>"CG-2"
```
it should say that the extra fields are called "maxspeed", "old_ref", "oneway", "ref" and "junction". This is useful for specifying the extra attributes. 

I'm thinking about some sort of regex but any suggestion is welcome.

### LINKED ISSUE #109: Change extra_attributes into extra_tags
See https://github.com/ITSLeeds/osmextract/issues/107#issuecomment-669981804

### PR DESCRIPTION
**Pull request by agila5**
Fix #80 
FIx #58
Fix #107

### CODE REVIEW

**Comment by agila5** [comment]
Hi @Robinlovelace If you want to check the PR I modified and documented a lot of new things, otherwise just merge!

**Comment by Robinlovelace** [comment]
Thanks @agila5 I'm checking it now...

File: NAMESPACE [hunk 1]
```diff
@@ -6,6 +6,7 @@ S3method(oe_match,numeric)
 S3method(oe_match,sfc_POINT)
 export(oe_download)
 export(oe_get)
+export(oe_get_keys)
```
Comment (Robinlovelace) [inline NAMESPACE:9]: :tada:

File: R/get.R [hunk 2]
```diff
@@ -38,18 +38,21 @@
...
+#'   [gdal.org](https://gdal.org/drivers/vector/osm.html). Check Details at
+#'   `oe_vectortranslate()`.
 #' @param extra_attributes Which addition columns, corresponding to OSM keys,
-#' should be in the resulting dataset? `NULL` by default.
-#' @param force_vectortranslate Force the original `.pbf` file to be translated
-#' into a `.gpkg` file, even if a `.gpkg` associated with the `provider` zone
-#' already exists.
+#'   should be in the resulting dataset? `NULL` by default. Check Details at
+#'   `oe_vectortranslate()` and `oe_get_keys()`.
+#' @param force_vectortranslate Boolean. Force the original `.pbf` file to be
```
Comment (Robinlovelace) [inline R/get.R:50]: Should say `FALSE` by default. I'll add that.

File: R/vectortranslate.R [hunk 3]
```diff
@@ -1,11 +1,92 @@
...
+#'
+#'   The parameter `vectortranslate_options` is used to control the arguments
+#'   that are passed to `ogr2ogr` via `sf::gdal_utils()` when converting between
+#'   `.pbf` and `.gpkg` formats. `ogr2ogr` can perform various operations during
+#'   the conversion process, such as spatial filters or SQL queries. These
+#'   operations are determined by the `vectortranslate_options` argument. If
+#'   `NULL` (default value), then `vectortranslate_options` is set equal to
+#'   `c("-f", "GPKG", "-overwrite", "-oo", paste0("CONFIG_FILE=", osmconf_ini),
+#'   "-lco", "GEOMETRY_NAME=geometry", layer)`. Explanation:
+#'   \itemize{
```
Comment (Robinlovelace) [inline R/vectortranslate.R:64]: I suggest we switch to Markdown style documentation: https://roxygen2.r-lib.org/articles/markdown.html

File: R/vectortranslate.R [hunk 4]
```diff
@@ -1,11 +1,92 @@
...
+#'       [Open Options](https://gdal.org/drivers/vector/osm.html#open-options)
+#'       for the `.osm.pbf` file and change the `CONFIG` file (in case the user
+#'        asks for any extra attribute or a totally different CONFIG file);
+#'        \item `"-lco", "GEOMETRY_NAME=geometry"` is used to change the
+#'        [layer creation options](https://gdal.org/drivers/vector/gpkg.html?highlight=gpkg#layer-creation-options)
+#'        for the `.gpkg` file and modify the name of the geometry column;
+#'        \item `layer` indicates which layer should be converted.
+#'   }
+#'   Check the introductory vignette, the help page of `sf::gdal_utils()` and
+#'   [here](https://gdal.org/programs/ogr2ogr.html) for an extensive
```
Comment (Robinlovelace) [inline R/vectortranslate.R:78]: Inconsistent: using markdown in some places and old school documentation syntax in others. That would be 

`\href{https://...}{hehe}` in old school docs syntax I think. No change needed on this line, just flagging.

File: R/vectortranslate.R [hunk 5]
```diff
@@ -16,36 +97,115 @@
...
-  gpkg_file_path = paste0(
-    # I need the double file_path_san_ext to cancel the .osm and the .pbf
-    tools::file_path_sans_ext(tools::file_path_sans_ext(file_path)),
-    ".gpkg"
-  )
+  # Check that the input file was specified using the format
+  # ".../something.pbf". This is important for creating the .gpkg file path.
+  if (tools::file_ext(file_path) != "pbf") {
+    stop(
+      "The input file must be specified using the appropriate extension, i.e. ",
```
Comment (Robinlovelace) [inline R/vectortranslate.R:132]: Could something more concise work better? For example

```
       "file_path must be a .pbf files" 
```

File: R/vectortranslate.R [hunk 6]
```diff
@@ -174,3 +332,75 @@ get_ini_layer_defaults = function(layer) {
...
+#' @inheritParams oe_get
+#' @param file_path The path of `.gpkg` file, typically created using
+#'   `oe_vectortranslate()` or `oe_get()`.
+#'
+#' @return A character vector indicating the name of all keys stored in
+#'   "other_tags" field.
+#' @export
+#'
+#' @examples
+#' itsleeds_gpkg <- oe_get("itsleeds", provider = "test", download_only = TRUE)
```
Comment (Robinlovelace) [inline R/vectortranslate.R:366]: Inconsistent style, we should use arrow assignment everywhere or nowhere.

**Comment by agila5** [comment]
Thanks for the review! Are you working on the changes? Otherwise, I will fix them tomorrow. I would use md syntax eveywhere in the docs and = assignment

**Comment by Robinlovelace** [comment]
> Are you working on the changes? 

No, please let me know tomorrow when it's good to ship. Thanks, good workflow.

File: tests/testthat/test-vectortranslate.R [hunk 7]
```diff
@@ -0,0 +1,42 @@
+# Prepare for the tests
+its_match = oe_match("ITS Leeds", provider = "test")
```
Comment (Robinlovelace) [inline tests/testthat/test-vectortranslate.R:2]: Great set of tests.

File: R/vectortranslate.R [hunk 8]
```diff
@@ -174,3 +341,75 @@ get_ini_layer_defaults = function(layer) {
...
+#'   `oe_vectortranslate()` or `oe_get()`.
+#'
+#' @return A character vector indicating the name of all keys stored in
+#'   "other_tags" field.
+#' @export
+#'
+#' @examples
+#' itsleeds_gpkg = oe_get("itsleeds", provider = "test", download_only = TRUE)
+#' oe_get_keys(itsleeds_gpkg)
+oe_get_keys <- function(
```
Comment (Robinlovelace) [inline R/vectortranslate.R:377]: Still a style issue.

**Pull request merged**